plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -885,6 +886,15 @@ public Optional<ConnectorOutputMetadata> finishCreateTable(
             metastore.createTable(session, table, principalPrivileges);
         }
         catch (Exception e) {
+            Optional<Table> existingTable = metastore.getTable(schemaName, tableName);
+            if (existingTable.isPresent()) {
```

Comment:
@aczajkowski @losipiuk @findepi

Reply:
This looks exactly as case we had in tests with Hive and MinIO. 
- call createTable
- call is broken due to Read timeout (but Hive continue to do the job and completes)
- we retry createTable (internal hive client implementation)
- we get TableAlreadyExists
This was addressed in test by extending Hive read timeout to 20s from default 10s.